bin/network-monitor/src/deploy/counter.rs

```diff
@@ -0,0 +1,72 @@
+//! Counter program account creation functionality.
+
+use std::path::Path;
+
+use anyhow::{Context, Result};
+use miden_lib::transaction::TransactionKernel;
+use miden_objects::account::{
+    Account,
+    AccountBuilder,
+    AccountComponent,
+    AccountFile,
+    AccountId,
+    AccountStorageMode,
+    AccountType,
+    StorageSlot,
+};
+use miden_objects::{Felt, Word};
+use tracing::instrument;
+
+use crate::COMPONENT;
+
+/// Create a counter program account with custom MASM script.
+#[instrument(target = COMPONENT, name = "create-counter-account", skip_all, ret(level = "debug"))]
+pub fn create_counter_account(owner_account_id: AccountId) -> Result<Account> {
+    // Load and customize the MASM script
+    let script =
+        include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/src/assets/counter_program.masm"));
+
+    // Compile the account code
+    let owner_felts: [Felt; 2] = owner_account_id.into();
+    let owner_word = Word::from([owner_felts[0], owner_felts[1], Felt::new(0), Felt::new(0)]);
+
+    let account_code = AccountComponent::compile(
+        script,
+        TransactionKernel::assembler(),
+        vec![StorageSlot::empty_value(), StorageSlot::Value(owner_word)],
+    )?
+    .with_supports_all_types();
+
+    const INCR_NONCE_AUTH_CODE: &str = r"
+        use.miden::account
+        use.std::sys
+
+        export.auth__basic
+            exec.account::incr_nonce
+            drop
+        end
+    ";
+
+    let incr_nonce_auth =
+        AccountComponent::compile(INCR_NONCE_AUTH_CODE, TransactionKernel::assembler(), vec![])
+            .context("failed to compile increment nonce auth component")?
+            .with_supports_all_types();
```

Comment:
I wonder if `IncrNonce` component from the mock tools from `miden-testing` can be used here instead (I think they are pretty much equivalent).